.github/workflows/cd-test-workflow.yml

```diff
@@ -0,0 +1,88 @@
+name: CD workflow
+run-name: ${{ github.actor }} is learning CD workflow in Github Actions
+on:
+  workflow_dispatch:
+    inputs:
+      flavor:
+        description: 'Choose the deployment environment'
+        required: true
+        default: 'staging'
+        type: choice
+        options:
+          - staging
+          - production
+
+jobs:
+  generate-apk-build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: cicd
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '21'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+        with:
+          gradle-version: 'wrapper'
+
+      - name: Decode Keystore
+        run: |
+          echo "${{ secrets.CD_KEYSTORE_FILE }}" | base64 --decode > app/cdapp.jks
+
+      - name: Generate ${{ github.event.inputs.flavor }} Release Apk
+        run: ./gradlew assemble${{ github.event.inputs.flavor }}Release
```

Comment:
Will `./gradlew` actually work if the project is in `cicd` directory instead of the repository's root directory?

Reply:
Ah, now I noticed 

```
    defaults:
      run:
        working-directory: cicd
```

so I guess all good 👍